src/app/clusters/soil-measurement-server/CodegenIntegration.cpp

```diff
@@ -0,0 +1,114 @@
+/*
+ *
+ *    Copyright (c) 2025 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include <app/clusters/soil-measurement-server/CodegenIntegration.h>
+#include <app/clusters/soil-measurement-server/soil-measurement-cluster.h>
+#include <app/static-cluster-config/SoilMeasurement.h>
+#include <app/util/config.h>
+#include <data-model-providers/codegen/ClusterIntegration.h>
+
+using namespace chip;
+using namespace chip::app;
+using namespace chip::app::Clusters;
+using namespace chip::app::Clusters::SoilMeasurement;
+using namespace chip::app::Clusters::SoilMeasurement::Attributes;
+
+namespace {
+
+constexpr size_t kSoilMeasurementFixedClusterCount = SoilMeasurement::StaticApplicationConfig::kFixedClusterConfig.size();
+constexpr size_t kSoilMeasurementMaxClusterCount   = kSoilMeasurementFixedClusterCount + CHIP_DEVICE_CONFIG_DYNAMIC_ENDPOINT_COUNT;
+
+LazyRegisteredServerCluster<SoilMeasurementCluster> gServer;
+
+const Globals::Structs::MeasurementAccuracyRangeStruct::Type kDefaultSoilMoistureMeasurementLimitsAccuracyRange[] = {
+    { .rangeMin = 0, .rangeMax = 100, .percentMax = MakeOptional(static_cast<chip::Percent100ths>(10)) }
+};
+
+const SoilMoistureMeasurementLimits::TypeInfo::Type kDefaultSoilMoistureMeasurementLimits = {
```

Comment:
I think these were intentionally NOT having an integration (sine https://github.com/project-chip/connectedhomeip/pull/41508) since these were used in a code-driven manner to start with.